### ISSUE TITLE
[BUG]: Error in examples/FinancialAnalysisSystem.ipynb

### ISSUE THREAD

**Issue by LuciAkirami** [raga-ai-hub/ragaai-catalyst#38]
# Bug Report

**Describe the Bug**
There is an error in the FinancialAnalysisSystem.ipynb file. In the `FinancialAnalysisSystem` class, the` financial_advisor_agent() ` function needs to contain `*args` and `**kwargs` in its parameters. Else we will get a  `TypeError: FinancialAnalysisSystem.financial_advisor_agent() got an unexpected keyword argument 'tool_call'`. Also, the last line should be `tracer.stop()` not `analysis_system.tracer.stop()`

**To Reproduce**
Steps or code snippets to reproduce the behavior, like:
```
1. Install AgentNeo using `pip install agentneo`
2. Run till "Running the Analysis" in the notebook
3. Observe the error or unexpected behavior.
```

**Expected Behavior**
```
Expected Output
Project 'financial_analysis_project40.5533834094087828' found.
HTTP Request: GET https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json "HTTP/1.1 200 OK"
Tracing Started.

HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 200 OK"
HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 200 OK"

Analysis for AAPL:
Stock Data: {'symbol': 'AAPL', 'price': 264.82, 'change': 2.83}
News Sentiment: 0.04545454545454545
Economic Indicators: {'gdp_growth': 3.62, 'unemployment_rate': 8.06, 'inflation_rate': 3.65}

Investment Recommendation:
Given the market outlook for Apple Inc. (AAPL) and the investor's moderate risk tolerance, a balanced investment recommendation would be in order. With the stock price of Apple Inc. having increased by 2.83% to $264.82, indicating positivity in its market performance, and a slight positive bias in news coverage, the company showcases resilience and potential for continued growth. Apple’s consistent performance, substantial market share in its industry, and ability to innovate contribute to its attractiveness as an investment.

### Investment Recommendation: 

**Consider a Partial Position with Diversification:**

1. **Buy into Apple Inc. (AAPL):** Considering the positive stock performance and news sentiment, initiating or increasing a position in Apple could be a wise move for a portfolio. Given the moderate risk tolerance, it might not be prudent to allocate a large portion of the portfolio to a single stock. Therefore, consider buying shares of Apple Inc. to constitute a significant, yet balanced, portion of the equity
Executing buy order for AAPL
Tracing Completed.
Data saved to the database and JSON file.
```

**Actual Behavior**
```
Project 'financial_analysis_project40.7691261821481966' found.
HTTP Request: GET https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json "HTTP/1.1 200 OK"
Tracing Started.
Error in agent 'FinancialAdvisorAgent': FinancialAnalysisSystem.financial_advisor_agent() got an unexpected keyword argument 'tool_call'
Traceback (most recent call last):
  File "/home/user/Desktop/projects/agentneo-test/test.py", line 141, in <module>
    analysis_system.run_analysis("AAPL", "moderate")
  File "/home/user/Desktop/projects/agentneo-test/test.py", line 115, in run_analysis
    recommendation = self.financial_advisor_agent(stock_symbol, risk_tolerance)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/site-packages/agentneo/tracer.py", line 412, in sync_wrapper
    return asyncio.run(async_wrapper(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/site-packages/agentneo/tracer.py", line 355, in async_wrapper
    result = await asyncio.to_thread(func, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/miniconda3/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: FinancialAnalysisSystem.financial_advisor_agent() got an unexpected keyword argument 'tool_call'
```
**Logs and Screenshots**
N/A

**Environment Details**
- **Operating System**: Ubuntu 20.04
- **Python Version**: 3.11.5
- **Node.js Version**: 21.6.2
- **npm/yarn Version**: 10.8.2
- **AgentNeo Version**: 1.0.3
- **Relevant Packages**: OpenAI 1.51.2

**AgentNeo Configuration**
Provide any custom configuration settings or code modifications:
```python
# adding *args and **kwargs fixes the code
def financial_advisor_agent(self, stock_symbol, risk_tolerance, *args, **kwargs):
...
# replace analysis_system.tracer.stop() with tracer.stop()
tracer.stop()
```

Can I raise a PR for this fix?

**Comment by rehan-ai** [comment]
can you try this with `agentneo==1.1.b2` , this comes from our V1.1 branch.
Let me know if the issue persist and in that case do raise a PR to fix this

**Comment by LuciAkirami** [comment]
Yup, the primary error(tool_call error) gets resolved in `agentneo=1.1.b2`. But the last code line still produces error. That is, the last line should be `tracer.stop()` not `analysis_system.tracer.stop()` as the `analysis_system` does not have any trace method. 

Error:
```
Traceback (most recent call last):
  File "/home/user/Desktop/projects/agentneo-test/testingv11.py", line 144, in <module>
    analysis_system.tracer.stop()
AttributeError: 'FinancialAnalysisSystem' object has no attribute 'tracer'
```

Fix:
Changing  `analysis_system.tracer.stop()`  to `tracer.stop()` 

Raising a PR to fix this
